src/mem/slab.h

```diff
@@ -121,12 +128,21 @@ namespace snmalloc
         return Superslab::NoSlabReturn;
       }
 
-      // Remove from the sizeclass list and dealloc on the superslab.
+      size_t count = 1;
+      // Check free list is well-formed.
```

Comment:
Expand on "well-formed"?  On `IntegerPointer` architectures with `CHECK_CLIENT`, WHP, we'll catch tampering with the list as well as cyclicity due to the conflicting previous addresses for the earliest node participating in the cycle.

Reply:
Probably should `ifdef` this code.